elsasserlib/R/normGeneCounts.R

```diff
@@ -0,0 +1,48 @@
+
+#' Quantile Normalization of Gene Count Matrix
+#'
+#' This function replaces original values with estimated values according 
+#' to the rank within each sample.
+#'
+#' @param gene.counts Gene count table for normalization
+#' @return Normalised Counts
+#' @export
+quantileNorm <- function(gene.counts)
+{ 
+  rank_mat <- apply(gene.counts, 2, rank, ties.method = "random")
+  means <- rowMeans(gene.counts)
+  rank_mean <- rank(means, ties.method = "random")
+  norm_mat <- apply(rank_mat, 2, function(x) means[x])
+  return(norm_mat)
+}
+
+#' Size Factor Normalization of Gene Count Matrix
+#'
+#' This function devides original values with size factors of each sample according 
+#' to DESeq method.
+#'
+#' @param gene.counts Gene count table for normalization
+#' @return Normalised Counts
+#' @export
+sizeFactorNorm <- function(gene.counts)
+{ 
+  geoMeans = apply(gene.counts, 1, function(x) exp(sum(log(x[x > 0]), na.rm=T) / length(x)) )
+  sf = apply(sweep(gene.counts, 1, geoMeans,'/'), 2, median)
+  return(sweep(gene.counts, 2, sf,'/'))
+}
+
+#' Median Ratio Normalization of Gene Count Matrix
+#'
+#' This function devides original values with median ratios comparing to the first sample 
+#' according to edgeR method.
+#'
+#' @param gene.counts Gene count table for normalization
+#' @return Normalised Counts
+#' @export
+mrnNorm <- function(gene.counts)
```

Comment:
here, I would have a more self-explanatory name for the function: `mean_ratio_norm`, unless it is some kind of naming standard?

Also, `{` at the same line.